### ISSUE TITLE
Вопрос по 3 лабе (18 вариант)

### ISSUE THREAD

**Issue by yuri-pechatnov** [agsh/mipt2016#16]
Такая проблема:
надо просмотреть 65 страниц списка разработчиков Microsoft (18 вариант). И в процессе меня сервер посылает с ошибкой - слишком много запросов. А задержку вставить никак не получается. Async.Sleep либо ставлю неправильно, либо он вообще тут не подходит(
Как с этим бороться?

upd:
Поставил считаться фибоначчи в промежутки и все равно ошибка от сервера( А оно и так уже долго работает

ошибка 429 с github 
<div class="container"> 

<h1>Whoa there!</h1> 

<p>You have triggered an abuse detection mechanism.<br> 
Please wait a few minutes before you try again.</p> 

<div id="suggestions">

**Comment by max-dankow** [comment]
Такая же проблема, весь вечер мучаюсь

**Comment by agsh** [comment]
Мне бы пример кода для редактирования

**Comment by max-dankow** [comment]
Вот упрощенный пример чтения страниц результатов поиска (локализованный баг)
[for i in [1..20] -> try HtmlDocument.Load("https://github.com/search?l=F%23&o=desc&p=" + i.ToString() + "&q=language%3AF%23&ref=advsearch&s=stars&type=Repositories&utf8=%E2%9C%93" ); i with _ -> -1]

**Comment by agsh** [comment]
У меня достаточно подождать 10 секунд после каждого запроса, чтобы всё прошло:

``` fsharp
open System.Threading
[for i in [1..20] -> try HtmlDocument.Load( "https://github.com/search?l=F%23&o=desc&p=" + i.ToString() + "&q=language%3AF%23&ref=advsearch&s=stars&type=Repositories&utf8=%E2%9C%93"); Thread.Sleep(10000); i with _ -> -1]
```

**Comment by max-dankow** [comment]
в моем задании нужно найти топ 500 в поиске  = 50 страниц по 10 репозиториев -> 50 \* 10 секунд - 8 минут где-то, долговато, как быть?Даже если уменьшить число, то все равно достаточно, не по тестировать( Может есть более оптимальный способ делать запросы в поисковой системе гитхаба..?

**Comment by max-dankow** [comment]
Да, спасибо, это работает корректно, но остается проблема времени

**Comment by agsh** [comment]
Можно взять html-код одной страницы для тестирования и использовать вместо метода `Load` метод `HtmlDocument.Parse`, который принимает на вход строку и строит такое же дерево
